utils/calendrical_calculations/src/chinese_based.rs

```diff
@@ -473,6 +473,65 @@ pub fn days_in_month<C: ChineseBased>(
     (result, next_new_moon)
 }
 
+/// Returns the last day of every month in the year, as well as a leap month index (1-indexed) if any
+/// In the case of no leap months, month 12 and 13 will have identical "last day"
+/// values
+pub fn month_structure_for_year<C: ChineseBased>(
+    new_year: RataDie,
+    next_new_year: RataDie,
+) -> ([RataDie; 13], Option<NonZeroU8>) {
+    let mut ret = [RataDie::new(0); 13];
+    ret[12] = next_new_year - 1; // last day of last month is the day before the next new year
+
+    let mut current_month_start = new_year;
+    let mut current_month_major_solar_term = major_solar_term_from_fixed::<C>(new_year);
+    let mut leap_month_index = None;
+    for i in 0u8..12 {
+        let next_month_start = new_moon_on_or_after::<C>((current_month_start + 28).as_moment());
```

Comment:
Question: does `28` work? I remember Reingold using various constants in different contexts

Reply:
Yeah, since on-or-after starts at midnight, so the potential off-by-one error is in the other direction in our favor. We know that next_month_start must be current_month_start + 29 or 30.